### PR TITLE
gateway-api: Bump the version to v1.0.0

### PR DESCRIPTION
**Pull request by sayboras**
### Description

Please refer to individual commit for more details. This is to prepare for v1.0.0 next week.

```release-note
Add support for Gateway API v1.0
```

### CODE REVIEW

**Review by rolinh — APPROVED** [review]
Vendor changes lgtm

File: Documentation/operations/upgrade.rst [hunk 1]
```diff
@@ -329,8 +329,12 @@ Annotations:
 * ``enable-endpoint-routes`` now automatically sets ``enable-local-node-route``
   to false, as local node routes are redundant when per-endpoint routes are enabled.
 * L7 visibility using Pod annotations (``policy.cilium.io/proxy-visibility``) is
-  no longer supported. 
+  no longer supported.
   We recommend users to switch to L7 policies instead (see :ref:`proxy_visibility`).
+* If you are using Gateway API, please make sure that that new v1 CRDs are installed.
```
Comment (learnitall) [inline Documentation/operations/upgrade.rst]: I think there's a duplicate `that` here.

```suggestion
* If you are using Gateway API, please make sure that new v1 CRDs are installed.
```

**Review by learnitall — CHANGES_REQUESTED** [review]
I found one typo, and I think there needs to be one more change in Documentation/network/servicemesh/gateway-api/gateway-api.rst` on line 53? 

```
More examples can be found in the `upstream repository <https://github.com/kubernetes-sigs/gateway-api/tree/v0.8.1/examples/standard>`_.
```

Otherwise LGTM!

**Review by meyskens — APPROVED** [review]
Technical changes look good :)

**Review by brlbil — APPROVED** [review]
CI changes lgtm.

File: Documentation/operations/upgrade.rst (hunk 1 above)
Comment (sayboras) [reply Documentation/operations/upgrade.rst]: thanks a lot :+1:

**Review by learnitall — APPROVED** [review]
Thanks!

**Pull request merged**